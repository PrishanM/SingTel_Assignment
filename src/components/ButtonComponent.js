import React from "react";
import { Text, StyleSheet, TouchableOpacity, Image } from "react-native";
import { appColors } from "../styles/colors";
import dimensions from "../styles/dimensions";

/**
 * Common Button Component
 * @param title
 * @param isDisabled
 * @param isOutlined
 * @param onPressButton
 * @param width
 * @param testId
 * @param height
 * @param icon
 * @returns {JSX.Element}
 * @constructor
 */
const ButtonComponent = ({
                           title,
                           isDisabled = false,
                           isOutlined = false,
                           onPressButton,
                           width = "65%",
                           testId = "",
                           height = dimensions.buttonHeight,
                           icon = ""
                         }) => {

  return (
    <TouchableOpacity testId={testId}
                      style={[buttonStyles.buttonParentView,
                        {
                          height: height,
                          width: width,
                          backgroundColor: isDisabled ? appColors.fontColor1 : isOutlined ? appColors.backgroundColor : appColors.primaryButtonColor,
                          borderWidth: isDisabled ? 0 : 1,
                        },
                      ]}
                      disabled={isDisabled}
                      onPress={() => {
                        if (!isDisabled) {
                          onPressButton(title);
                        }
                      }}>

      {
        icon &&
        <Image source={icon}
               style={{ height: dimensions.buttonIconSize, width: dimensions.buttonIconSize, marginRight: dimensions.margin4 }} />

      }


      <Text
        style={[buttonStyles.buttonTextStyle, { color: isDisabled ? appColors.colorWhite : isOutlined ? appColors.primaryButtonColor : appColors.colorWhite }]}>
        {title}
      </Text>

    </TouchableOpacity>
  );

};

export default ButtonComponent;

const buttonStyles = StyleSheet.create({
  buttonParentView: {
    borderRadius: dimensions.borderRadius,
    borderWidth: 1,
    borderColor: appColors.primaryButtonColor,
    marginVertical: 15,
    justifyContent: "center",
    alignItems: "center",
    flexDirection: "row",
  },
  buttonTextStyle: {
    fontSize: 14,
  },
});
