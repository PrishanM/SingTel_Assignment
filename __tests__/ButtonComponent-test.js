import React from 'react';
import renderer from 'react-test-renderer';
import ButtonComponent from "../src/components/ButtonComponent";

test('renders correctly', () => {
  const tree = renderer.create(<ButtonComponent />).toJSON();
  expect(tree).toMatchSnapshot();
})
