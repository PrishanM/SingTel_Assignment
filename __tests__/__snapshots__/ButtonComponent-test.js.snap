// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EB736D",
      "borderColor": "#EB736D",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 54,
      "justifyContent": "center",
      "marginVertical": 15,
      "opacity": 1,
      "width": "65%",
    }
  }
>
  
  <Text
    style={
      Array [
        Object {
          "fontSize": 14,
        },
        Object {
          "color": "#FFFFFF",
        },
      ]
    }
  />
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EB736D",
      "borderColor": "#EB736D",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 54,
      "justifyContent": "center",
      "marginVertical": 15,
      "opacity": 1,
      "width": "65%",
    }
  }
>
  
  <Text
    style={
      Array [
        Object {
          "fontSize": 14,
        },
        Object {
          "color": "#FFFFFF",
        },
      ]
    }
  />
</View>
`;
